### PR TITLE
Feature/skills sh compatibility

### PR DESCRIPTION
**Pull request by kylejryan**
# Skills.sh-Compatible Skill Support

Adds skills.sh-compatible skill support—skills are discovered from `~/.pensar/skills/` directories (each containing a `skill.md` manifest), parsed into a unified registry, and surfaced throughout the app.

### Skill Scanner & Parser

New `scanner.ts` discovers skill directories, `parser.ts` extracts frontmatter metadata (name, description, version, tags) and instruction content from `skill.md` files.

### Unified Registry

`registry.ts` provides a single source of truth (`SkillsRegistry`) that replaces the old ad-hoc skill loading. Skills can be enabled, activated/deactivated, and queried by slug.

### Agent Tools

Added `useSkill`, `deactivateSkill`, and `listSkills` tools so the offensive security agent can activate and manage skills during a session.

### Operator Dashboard Integration

Skills appear in operator mode autocomplete (alongside `/models` and `/skills` commands). Inline `/skill-slug` references in prompts are detected, activated, and injected into the system prompt. Skill instructions are appended as active skill context.

### Home Screen Cleanup

Home screen autocomplete only shows built-in commands—skill slugs are excluded to keep it clean. Skills are browsable via `/skills` and activatable in operator mode.

### Slash Command Highlighting

`PromptInput` now highlights `/slug` patterns in the input text when they match known skills.

### Skills Dialog

Replaced the old `create-skill-wizard.tsx` with a new `skills-dialog.tsx` for browsing installed skills.

### Prompt Input Improvements

Extracted logic into `prompt-input-logic.ts` with inline slash detection, tab completion for mid-text `/slug` tokens, and comprehensive tests.

### Tests

- `bun test src/core/skills/parser.test.ts` — manifest parsing (frontmatter extraction, edge cases)
- `bun test src/core/skills/registry.test.ts` — registry enable/activate/deactivate/refresh lifecycle
- `bun test src/core/skills/scanner.test.ts` — directory scanning and discovery
- `bun test src/tui/components/operator-dashboard/logic.test.ts` — autocomplete filtering, command routing, inline skill extraction
- `bun test src/tui/components/shared/prompt-input-logic.test.ts` — inline slash detection, tab completion, suggestion filtering
- `bun run tsc --noEmit` — clean (only pre-existing errors in unrelated `scripts/test-auth-browser.ts`)

### Manual Testing

Verified `/` autocomplete on home screen shows only built-in commands, operator mode shows commands + skill slugs, skill activation injects instructions into system prompt.

---

> [!NOTE]
> **Medium Risk**
> Introduces a new on-disk skills discovery/parsing pipeline and threads it through operator prompt/tooling, which can affect agent behavior and command UX if skill scanning or prompt composition is incorrect.
> 
> **Overview**
> Adds skills.sh-compatible skill support via a new `SkillsRegistry` that scans multiple roots (project + user), parses `SKILL.md` YAML frontmatter, deduplicates by slug, and loads full instructions on demand.
> 
> Integrates skills into the offsec agent by passing an optional `skillsRegistry` through tool context and conditionally enabling a new `read_skill` tool (with `SKILL_TOOL_NAMES`), plus appending a generated skills catalog into the operator system prompt.
> 
> Updates the TUI to use the registry for autocomplete and skill execution, replaces `/create-skill` with a `/skills` browser dialog (with detail view + on-demand instruction loading), and enhances `PromptInput` with inline `/slug` detection, mid-text tab completion, and optional slash-command highlighting (with new tests).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 13011b9e3d3c88af19196a14f87542fcb9574002. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/tui/components/operator-dashboard/index.tsx [hunk 1]
```diff
@@ -771,17 +791,37 @@ export default function OperatorDashboard({
         },
       } satisfies ConsumeCallbacks;
 
-      const commonInput = {
```
[the remark this answers is not in the document]
Comment (kylejryan) [reply src/tui/components/operator-dashboard/index.tsx:821]: @cursoragent Fix this

**Comment by kylejryan** [comment]
@joshkotrous Resolved the conflicts here

**Review by jorgeraad — COMMENTED** [review]
Niceee, sick feature. I think the foundation looks good like the skills parsing and integration with the TUI. I do have a few comments on the design:
- I'd avoid introducing a distinction between legacy vs new skills systems. If we introduce one consolidated system, we can avoid tech debt that can be tough to remove later.
- No activate vs deactivate skill tools - these eliminate the benefits of prompt caching (since we'd be modifying the system prompt each time), adds complexity, and doesn’t take advantage of skills’ built in advantage of progressive disclosure. Would be simpler + likely better performance to follow Anthropic's lead. Also looked at OpenCode and Codex as examples:
  - https://claude.com/blog/equipping-agents-for-the-real-world-with-agent-skills
  - https://deepwiki.com/search/explain-to-me-how-skills-work_9f144a95-8935-49f6-b289-f95c3a8cb313?mode=fast
  - https://deepwiki.com/search/explain-to-me-how-skills-work_3bcfaafb-7b6d-4931-91c4-0681abf4ca69?mode=fast
- Instead, simpler solution: list of skills frontmatters loaded into the prompt once and agent only has one tool to load the skill when it needs to use it. Can be returned in the tool result. An alternative we can consider would be to auto detect skills and simply inject the skills content via code. That said, requires more code and there could more complicated to handle edge cases.
  - downside of no list-skills tool is the user would need to start a new session or point the agent to the skill file location for the agent to use it. I think this is a fine tradeoff to avoid adding another tool as we should minimize the number of tools.

**Comment by kylejryan** [comment]
@jorgeraad Implemented your feedback, lmk what you think 🫡

File: src/tui/components/operator-dashboard/index.tsx [hunk 2]
```diff
@@ -1011,7 +1025,11 @@ export default function OperatorDashboard({
             approvalGateRef.current.updateConfig({ requireApproval: false });
             setOperatorState((prev) => ({ ...prev, requireApproval: false }));
           }
-          handleSubmit(action.content);
+          // Skill is already activated in the registry (system prompt).
+          // Send a brief message so the agent acknowledges and asks for a target.
+          handleSubmit(
```
Comment (jorgeraad) [inline src/tui/components/operator-dashboard/index.tsx]: Let's either expand skill to have the content, maybe with start and end tags, or directly encourage the agent to use the skill tool.

Let's remove the unnecessary question since the skill can encourage the agent to ask for more info if  that is the intention. This would alter the behavior of the skill.

File: src/core/skills/registry.test.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
+import { describe, it, expect, beforeEach, afterEach } from "vitest";
+import os from "os";
+import path from "path";
+import fs from "fs/promises";
+import { SkillsRegistry } from "./registry";
+
+const SKILLS_DIR = path.join(os.homedir(), ".pensar", "skills");
```
Comment (jorgeraad) [inline src/core/skills/registry.test.ts]: This touches the real skills dir. Use something like ` os.tmpdir()`.

**Review by jorgeraad — COMMENTED** [review]
Ok this looks great. You mind adding a screen recording showing what this looks like?

File: src/core/skills/registry.test.ts (hunk 3 above)
Comment (kylejryan) [reply src/core/skills/registry.test.ts]: @cursoragent Resolve this

Comment (kylejryan) [reply src/core/skills/registry.test.ts]: @jorgeraad Resolved in [a80fa4c](https://github.com/pensarai/apex/pull/308/commits/a80fa4cb8c191880a02066adf47e15dc30419ae7)

File: src/tui/components/operator-dashboard/index.tsx (hunk 2 above)
Comment (kylejryan) [reply src/tui/components/operator-dashboard/index.tsx]: @jorgeraad Resolve in [216fde7](https://github.com/pensarai/apex/pull/308/commits/216fde7f5ca88af2d867827c0a72eeba9ce2964a)

**Comment by kylejryan** [comment]
> Ok this looks great. You mind adding a screen recording showing what this looks like?
@jorgeraad See attached

https://github.com/user-attachments/assets/ef38769a-202c-437c-8614-58ce2c4adec7

**Review by jorgeraad — APPROVED** [review]
